products/extensions/subscriptions/guides/separate-checkout.md

```diff
@@ -0,0 +1,209 @@
+    <option value="<subscription-interval-id>">Yearly interval</option>
+  </select>
+</form>
+```
+
+</Tab>
+
+<Tab title="Headless (curl)">
+
+Store-API requests need the subscription headers to be set, see [Request scoping](#request-scoping).
+
+The only exception is adding a line item:
+
+```sh
+curl -XPOST '/store-api/suscription/checkout/line-item/add' -d '{
+    "lineItems": [{
+      "id": <product-id>,
+      ...
+    }],
+    "subscription-plan-option": <subscription-plan-id>,
+    "subscription-plan-option-<subscription-plan-id>-interval": <subscription-interval-id>
+  }'
+```
+
+</Tab>
+</Tabs>
+
+## Events
+
+Most of the events triggered within subscription checkout are prefixed with `subscription.`.
+These events are identical to normal checkout events.
+If you wish to use these events, you need to subscribe to them.
+A list of known prefixed events can be found in `Subscription/Framework/Event/SubscriptionEventRegistry.php`
+
+```php
+// Normal Event Listener
+class MyEventSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [CheckoutOrderPlacedCriteriaEvent::class => 'onOrderPlacedCriteria'];
+    }
+
+    public function onOrderPlacedCriteria(CheckoutOrderPlacedCriteriaEvent $event): void
+    {
+        // Your event handler logic
+    }
+}
+
+// Subscription Event Listener
+class MyEventSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return ['subscription.' . CheckoutOrderPlacedCriteriaEvent::class => 'onOrderPlacedCriteria'];
+    }
+
+    public function onOrderPlacedCriteria(CheckoutOrderPlacedCriteriaEvent $event): void
+    {
+        // Your event handler logic
+    }
+}
+```
+
+## Request scoping
+
+In Storefront, there is an additional URL parameter (`subscriptionToken`) that gets resolved.
+In headless, there are two header parameters that need to be set namely `sw-subscription-plan` and `sw-subscription-interval`.
+
+Below is an example of the context set on a subscription cart in the Storefront:
+
+```xml
+<route id="frontend.subscription.checkout.cart.page"
+        path="/subscription/checkout/cart/{subscriptionToken}"
+        methods="GET"
+        controller="subscription.storefront.controller.checkout::cartPage">
+    <default key="_noStore">true</default>
+    <default key="_routeScope"><list><string>storefront</string></list></default>
+    <default key="_subscriptionCart">true</default>
+    <default key="_subscriptionContext">true</default>
+    <default key="_controllerName">checkout</default>
+    <default key="_controllerAction">cartpage</default>
+    <default key="_templateScopes">subscription</default>
+    <option key="seo">false</option>
+</route>
+```
+
+And, here is an example of the headers set on a subscription cart using headless:
+
+```sh
+curl -XPOST '/store-api/subscription/{subscriptionId}/activate' /
```

Comment:
I know, this is only copied, but it is a really bad example, as during the activation of a subscription, there is no header to set.


```suggestion
curl -XPOST '/store-api/subscription/checkout/cart/line-item' /
```